### DISCUSSION TITLE
Feature request: rendering scaling.

### DISCUSSION THREAD

**Discussion by Daniel-Chin** [FakeFishGames/Barotrauma#12552] [Ideas and suggestions]
## Definition
Rendering scaling is a settings parameter, `k`, in [0, 1]. The GUI of the game will render at the game resolution, `r`. The 2D game world of Europa is then drawn at resolution `r * k`.  

## Typical usage
The player will set game resolution `r` to the monitor native resolution and set `k < 1`. 

## Motivations
- It allows preserving crispy GUI texts while reducing rendering computation load. 
- Many other modern games have such a setting. Tools to implement rendering scaling are probably well supported. 

## My case
- I have a 4K resolution laptop with an integrated GPU. If I put the game in 4K, I get 5 FPS. Impossible to fight boarders. If I lower the resolution, the framerate goes to 60/s, but the texts become blurry. 

Praise the honkmother. Peace.

**Comment by jigardi** [comment]
I like this suggestion a lot.  Monitors are moving to higher resolutions over time, this would help those who already have high-res monitors, but more importantly it would prevent the problem from worsening over time.
